### PR TITLE
Severity

### PR DESCRIPTION
**Pull request by sstrubberg**
### Brief

As a maintainer of the Core Components of Carbon, I would argue that we have an incredibly polished process for triaging defects in our codebase and turning those bugs around into fixes in our regular sprint cadence. 

However, I think things tend to fall apart when we extend the idea of triaging and fixing defects in the other pillars of our design system, our Figma kit and our Documentation. This PR aims to remedy that with a comprehensive point-of-view on our approach to applying severity to not only our Code, but our Kit and Docs as well.

### Testing

To review, click on the `Files changed` tab above, then click the button to display rich diff.

image

Scroll down to the Triaging section to see the table.

I'm looking for feedback on not only these descriptions, but on the very idea of assigning severity to defects in not just our code, but our docs and our kits as well.

### CODE REVIEW

**Comment by aagonzales** [comment]
Initial thought on docs, I think we could break down "missing" into major and minor pieces.  

_Docs Sev 2_: "The documentation is wrong or missing **major details**".

_Docs Sev 3_: "The documentation is mis-leading, somewhat true, **or missing minor details**"
- It does **not** impact the users understanding of how to use the component. Something like maybe a token was missed on the style tab.

File: docs/guides/support.md [hunk 1]
```diff
@@ -346,16 +346,12 @@ business impact that the feature will bring with future research done.
...
+| [1](https://github.com/carbon-design-system/carbon/labels/severity%3A%201) | Affects major functionality. Blocks users from completing tasks, exposes intellectual property or damages the brand. No workaround. | The kit is has major functionality defects, designers can’t design and there is no work-around                   | The documentation damages the brand or exposes intellectual property.                                                                          | The code has major functionality defects, breaking builds and can’t be worked around.                            | If the problem is in production, it must be fixed as soon as possible in a patch.                                                                                                |
```
Comment (tay1orjones) [inline docs/guides/support.md]: If you remove the markdown formatting around these links GitHub will display the actual label itself. It's still a link, too.

```suggestion
| https://github.com/carbon-design-system/carbon/labels/severity%3A%201 | Affects major functionality. Blocks users from completing tasks, exposes intellectual property or damages the brand. No workaround. | The kit is has major functionality defects, designers can’t design and there is no work-around                   | The documentation damages the brand or exposes intellectual property.                                                                          | The code has major functionality defects, breaking builds and can’t be worked around.                            | If the problem is in production, it must be fixed as soon as possible in a patch.                                                                                                |
```

File: docs/guides/support.md [hunk 2]
```diff
@@ -346,16 +346,12 @@ business impact that the feature will bring with future research done.
...
+| [2](https://github.com/carbon-design-system/carbon/labels/severity%3A%202) |                     Affects major functionality. Does not block users from completing tasks. Has a workaround.                      | The kit has a major functionality defect, but it doesn't block users from completing tasks. It has a workaround. | The documentation is wrong or missing major details. It conflicts with what we are saying in another part of the system.                       | The code has a major functionality defect, but it doesn't block users from completing tasks. It has a workaround | A release will be moved into Production with defects at this severity level included. These defects will have priority in future sprints to be resolved ASAP.                    |
```
Comment (tay1orjones) [inline docs/guides/support.md]:
```suggestion
| https://github.com/carbon-design-system/carbon/labels/severity%3A%202 |                     Affects major functionality. Does not block users from completing tasks. Has a workaround.                      | The kit has a major functionality defect, but it doesn't block users from completing tasks. It has a workaround. | The documentation is wrong or missing major details. It conflicts with what we are saying in another part of the system.                       | The code has a major functionality defect, but it doesn't block users from completing tasks. It has a workaround | A release will be moved into Production with defects at this severity level included. These defects will have priority in future sprints to be resolved ASAP.                    |
```

File: docs/guides/support.md [hunk 3]
```diff
@@ -346,16 +346,12 @@ business impact that the feature will bring with future research done.
...
+| [3](https://github.com/carbon-design-system/carbon/labels/severity%3A%203) |                     Affects minor functionality. Does not block users from completing tasks. Has a workaround.                      | The kit has a minor functionality defect. It doesn't block users from completing tasks. Has a work around.       | The documentation is mis-leading, somewhat true, or missing minor details. It doesn't present conflicts with other parts of the design system. | The code has a minor functionality defect. It doesn't block users from completing tasks. Has a work around.      | A release will be moved into Production with defects at this severity level included. These defects will have priority in future sprints after sev2's.                           |
```
Comment (tay1orjones) [inline docs/guides/support.md]:
```suggestion
| https://github.com/carbon-design-system/carbon/labels/severity%3A%203 |                     Affects minor functionality. Does not block users from completing tasks. Has a workaround.                      | The kit has a minor functionality defect. It doesn't block users from completing tasks. Has a work around.       | The documentation is mis-leading, somewhat true, or missing minor details. It doesn't present conflicts with other parts of the design system. | The code has a minor functionality defect. It doesn't block users from completing tasks. Has a work around.      | A release will be moved into Production with defects at this severity level included. These defects will have priority in future sprints after sev2's.                           |
```

File: docs/guides/support.md [hunk 4]
```diff
@@ -346,16 +346,12 @@ business impact that the feature will bring with future research done.
...
+| [4](https://github.com/carbon-design-system/carbon/labels/severity%3A%204) |                                        Unrelated to a user task. Does not need a workaround.                                        | Unrelated to a user task. Does not need a workaround.                                                            | Unrelated to a user task. Does not need a workaround.                                                                                          | Unrelated to a user task. Does not need a workaround.                                                            | A release will be moved into Production with defects at this severity level included. These will often not be addressed by the maintainer team as a priority in future releases. |
```
Comment (tay1orjones) [inline docs/guides/support.md]:
```suggestion
| https://github.com/carbon-design-system/carbon/labels/severity%3A%204 |                                        Unrelated to a user task. Does not need a workaround.                                        | Unrelated to a user task. Does not need a workaround.                                                            | Unrelated to a user task. Does not need a workaround.                                                                                          | Unrelated to a user task. Does not need a workaround.                                                            | A release will be moved into Production with defects at this severity level included. These will often not be addressed by the maintainer team as a priority in future releases. |
```

**Review by tay1orjones — APPROVED** [review]
Looks good! One minor suggestion on the first column

**Review by aagonzales — APPROVED** [review]
I think these are good for general descriptions. I think at some point we could better define what is considered "major" versus "minor"

**Comment by sstrubberg** [comment]
> I think these are good for general descriptions. I think at some point we could better define what is considered "major" versus "minor"

Agreed. In my mind, as we start to tackle these types of issues, we'll start to find harden our POV on major vs. minor. What's important is that we identifying defects and having the conversations.

**Review by alisonjoseph — APPROVED** [review]
❤️

File: docs/guides/support.md [hunk 5]
```diff
@@ -346,16 +346,12 @@ business impact that the feature will bring with future research done.
...
+| https://github.com/carbon-design-system/carbon/labels/severity%3A%201 | Affects major functionality. Blocks users from completing tasks, exposes intellectual property or damages the brand. No workaround. | The kit is has major functionality defects, designers can’t design and there is no work-around                   | The documentation damages the brand or exposes intellectual property.                                                                          | The code has major functionality defects, breaking builds and can’t be worked around.                            | If the problem is in production, it must be fixed as soon as possible in a patch.                                                                                                |
```
Comment (elycheea) [inline docs/guides/support.md]:
```suggestion
| https://github.com/carbon-design-system/carbon/labels/severity%3A%201 | Affects major functionality. Blocks users from completing tasks, exposes intellectual property or damages the brand. No workaround. | The kit is has major functionality defects, designers can’t design and there is no workaround                   | The documentation damages the brand or exposes intellectual property.                                                                          | The code has major functionality defects, breaking builds and can’t be worked around.                            | If the problem is in production, it must be fixed as soon as possible in a patch.                                                                                                |
```

File: docs/guides/support.md [hunk 6]
```diff
@@ -346,16 +346,12 @@ business impact that the feature will bring with future research done.
...
+| https://github.com/carbon-design-system/carbon/labels/severity%3A%203 |                     Affects minor functionality. Does not block users from completing tasks. Has a workaround.                      | The kit has a minor functionality defect. It doesn't block users from completing tasks. Has a work around.       | The documentation is mis-leading, somewhat true, or missing minor details. It doesn't present conflicts with other parts of the design system. | The code has a minor functionality defect. It doesn't block users from completing tasks. Has a work around.      | A release will be moved into Production with defects at this severity level included. These defects will have priority in future sprints after sev2's.                           |
```
Comment (elycheea) [inline docs/guides/support.md]:
```suggestion
| https://github.com/carbon-design-system/carbon/labels/severity%3A%203 |                     Affects minor functionality. Does not block users from completing tasks. Has a workaround.                      | The kit has a minor functionality defect. It doesn't block users from completing tasks. Has a workaround.       | The documentation is mis-leading, somewhat true, or missing minor details. It doesn't present conflicts with other parts of the design system. | The code has a minor functionality defect. It doesn’t block users from completing tasks. Has a workaround.      | A release will be moved into Production with defects at this severity level included. These defects will have priority in future sprints after sev2’s.                           |
```

**Review by elycheea — COMMENTED** [review]
I apologize for nitpicking. We have some inconsistent usage of `workaround`, `work-around`, and `work around`. 😉

**Comment by sstrubberg** [comment]
> I apologize for nitpicking. We have some inconsistent usage of `workaround`, `work-around`, and `work around`. 😉

no apology needed! Fixing now!

**Review by elycheea — APPROVED** [review]

**Review by laurenmrice — APPROVED** [review]
Thanks for doing this, Scott!

**Review by oliviaflory — APPROVED** [review]
Thank you Scott!

**Pull request merged**